### PR TITLE
Movie view design

### PR DESCRIPTION
**Pull request by MorsySoker**
Add MoviesView 
Connected ProfileView with the NavigationBar 
Add dismiss button to ProfileView
Can't Log in with essam -- root  
show password button doesn't work
don't refactor anything plz just leave notes

### CODE REVIEW

File: iMovie/Scenes/Movies/MoviesView.swift [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    .background(.clear)
+
+                Spacer()
+
+                ContentPickerView()
+            }
+        }
+        .ignoresSafeArea(.all, edges: .top)
+    }
+}
```
Comment (essamMohamedFahmi) [inline iMovie/Scenes/Movies/MoviesView.swift:38]: Format this file, recheck spaces

File: iMovie/Scenes/Movies/MoviesView.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//  iMovie
+//
+//  Created by MorsyElsokary on 08/01/2022.
+//
+
+import SwiftUI
+
+struct MoviesView: View {
+
+    // MARK: - Private Methods
```
Comment (essamMohamedFahmi) [inline iMovie/Scenes/Movies/MoviesView.swift:12]: We put methods at the bottom, not on the top

File: iMovie/Views/ContentPickerView.swift [hunk 3]
```diff
@@ -0,0 +1,148 @@
+//
+//  ContentPickerView.swift
+//  iMovie
+//
+//  Created by MorsyElsokary on 08/01/2022.
+//
+
+import SwiftUI
+
+enum ContentCriteria: String, CaseIterable {
```
Comment (essamMohamedFahmi) [inline iMovie/Views/ContentPickerView.swift:10]: Awesome but this should be called "MovieCategory" or something like that

File: iMovie/Views/ContentPickerView.swift [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+            currentSelector()
+        }
+        .background(.clear)
+    }
+
+    // MARK: - Private Methods
+
+    @ViewBuilder
+    private func currentSelector() -> some View {
```
Comment (essamMohamedFahmi) [inline iMovie/Views/ContentPickerView.swift:55]: Good separation but cases are very large, try separate cases in other methods ✌🏽

File: iMovie/Views/NavigationBarView.swift [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+struct NavigationBarView: View {
+
+    // MARK: - Properties
+
+    @State private var logoAnimated: Bool = false
+    @State private var showProfile: Bool = false
+
+    // MARK: - Body
+
+    var body: some View {
```
Comment (essamMohamedFahmi) [inline iMovie/Views/NavigationBarView.swift:19]: Divide these views into computed properties

**Review by essamMohamedFahmi — CHANGES_REQUESTED** [review]

File: iMovie/Views/NavigationBarView.swift [hunk 6]
```diff
@@ -8,55 +8,69 @@
...
+            
+            Spacer()
+            
+            showProfileButton
+        }
+    }
+    
+    // MARK: - Private Properties
+    
+    private var hamburgerMenuButton: some View {
```
Comment (essamMohamedFahmi) [inline iMovie/Views/NavigationBarView.swift:35]: Please look at login module and see how we create view computed properties

Comment (essamMohamedFahmi) [reply iMovie/Views/NavigationBarView.swift:35]: We put them above body and mark is View Components please just follow same structure

File: iMovie/Views/CategoryPickerView.swift [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+            currentSelector()
+        }
+        .background(.clear)
+    }
+
+    // MARK: - Private Methods
+
+    @ViewBuilder
+    private func displayContent() -> some View {
```
Comment (essamMohamedFahmi) [inline iMovie/Views/CategoryPickerView.swift:55]: Why this is method not computed property and also caller method should be before called one - journal style clean code

**Review by essamMohamedFahmi — CHANGES_REQUESTED** [review]

**Comment by essamMohamedFahmi** [comment]
Do not resolve things, just push edits

File: iMovie/Views/NavigationBarView.swift (hunk 5 above)
Comment (MorsySoker) [reply iMovie/Views/NavigationBarView.swift:19]: done

File: iMovie/Scenes/Movies/MoviesView.swift [hunk 8]
```diff
@@ -9,6 +9,14 @@ import SwiftUI
 
 struct MoviesView: View {
 
+    // MARK: - Properties
+
+    var topSafeAreaInsets: CGFloat {
```
Comment (essamMohamedFahmi) [inline iMovie/Scenes/Movies/MoviesView.swift:14]: Does this need to be a computed property and get calculated every time or it needs to be calculated the first time only?

**Review by essamMohamedFahmi — APPROVED** [review]

**Comment by essamMohamedFahmi** [comment]
Please squash and merge with the title "Add movies view"

**Pull request merged**